### DISCUSSION TITLE
Problem in running Zundel example

### DISCUSSION THREAD

**Discussion by rajorshichat** [MarsalekGroup/aml#7] [Q&A] [answered]
Hello,
I am trying to run the example simulation on zundel. However when I run the run-QbC.py code, I get an error that says that 'No such file or directory: 'iteration-001/train-000/scaling.data' in the first QbC iteration. When I check the iteration-001/train-000 directory I see no scaling.data file there. I see a 'nnp-scaling-stdout.err' file which says that 'opal_init' has failed. Can you help me solve this problem ?

**Comment by cschran** [comment]
Hi rajorshichat,

this sounds like an issue with your n2p2 setup.
Make sure that the functions `nnp-scaling` and `nnp-train` are available from the command line (the nnp/bin directory needs to be part of your path).
If that's not the reason, could you provide the output created in `iteration-001/train-000`?

Best,
Christoph

**Comment by rajorshichat** [comment]
Hello,
Thanks a lot for your reply.

Following your suggestion, I put the n2p2/bin folder (containing executable for nnp-train, nnp-scaling) into my path variable (/usr/local/bin). But I still get the same errors. I have attached the image of the error.
![AML_error](https://user-images.githubusercontent.com/28429566/145037920-c82f27f4-c1e4-4382-b159-f8f5a37c8ccd.png)

Within the iteration-001/train-000 directory, following files are generated - input.data, input.nn, nnp-train-stdout.log (blank file), nnp-scaling-stdlog.log (blank file) and two .err files. I contents of the .err files are - 
[mpiexec@rajorshi-HP-Pavilion-Notebook] match_arg (utils/args/args.c:159): unrecognized argument cpu-set
[mpiexec@rajorshi-HP-Pavilion-Notebook] HYDU_parse_array (utils/args/args.c:174): argument matching returned error
[mpiexec@rajorshi-HP-Pavilion-Notebook] parse_args (ui/mpich/utils.c:1597): error parsing input array
[mpiexec@rajorshi-HP-Pavilion-Notebook] HYD_uii_mpx_get_parameters (ui/mpich/utils.c:1649): unable to parse user arguments
[mpiexec@rajorshi-HP-Pavilion-Notebook] main (ui/mpich/mpiexec.c:149): error parsing parameters

**Comment by cschran** [comment, marked as answer]
Your error output indicates problems with mpi. You are using `mpich`, which seems to have trouble with our default MPI-command used to initiate the training. 
A quick workaround should be to change this line https://github.com/MarsalekGroup/aml/blob/f8adf1a7f81107e67a904f242142417f73265f98/aml/mlp.py#L364
to 
``` 
mode = 'MPI'
```

Other recommendations would be to check that n2p2 was compiled with `mpich` and that the number of cores accessible to the QbC run does not exceed `n_tasks*n_core_task`, which is 24 cores in the example.

Another workaround is to set `n_core_task=1`, which would trigger a serial n2p2 run. You can still parallelize up to the number of committee members with `n_tasks`.

**Comment by cenwanglai** [comment]
Hi Christoph

I have set the n_core_task = 1 in the example/02/run-Qbc.py code and run it directly with ./run-Qbc.py. It did not work and the erros like:
![fa2001b696711c5f79960f0f84f0811](https://user-images.githubusercontent.com/29187935/208883939-1df7a2fe-bf5b-4b22-99ff-0f4d113959e6.png)

What should I do then?

**Reply by cenwanglai** [reply]
Yes, I fixed the problem. Actually, the n2p2 should be compiled first. Are there any information included in the README file of AML for the dependency on n2p2?

**Reply by adilalam45** [reply]
can you please explain what exactly you did?
